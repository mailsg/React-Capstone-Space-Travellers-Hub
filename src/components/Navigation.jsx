import React from 'react';
import { NavLink } from 'react-router-dom';
import '../App.css';
import Nav from 'react-bootstrap/Nav';
import Stack from 'react-bootstrap/Stack';
import logo from '../images/logo.png';

const Navigation = () => (
  <Stack direction="horizontal" gap={3} className="mx-5 my-3 pb-3" style={{ borderBottom: '1px solid black' }}>
    <img src={logo} style={{ width: 65, height: 65 }} alt="logo" />
    <h1>Space Travelers` Hub</h1>
    <Nav className="ms-auto">
      <Nav.Item><Nav.Link as={NavLink} to="/" className="nav-link-effect">Rockets</Nav.Link></Nav.Item>
      <Nav.Item>
        <Nav.Link
          as={NavLink}
          className="nav-link-effect"
          to="/missions"
          style={{ marginRight: '1rem' }}
        >
          Missions
        </Nav.Link>
      </Nav.Item>
      <h6 style={{ marginTop: '0.5rem', marginRight: '1.5rem' }}>|</h6>
      <Nav.Item><Nav.Link as={NavLink} className="nav-link-effect" to="/myprofile" style={{ marginRight: '3.5rem' }}>My Profile</Nav.Link></Nav.Item>
    </Nav>
  </Stack>
);

export default Navigation;
