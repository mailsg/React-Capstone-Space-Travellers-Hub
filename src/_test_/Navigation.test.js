import '@testing-library/jest-dom';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Navigation from '../components/Navigation';

describe('User Interactions test', () => {
  it('will test user interaction', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Navigation />
        </MemoryRouter>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
