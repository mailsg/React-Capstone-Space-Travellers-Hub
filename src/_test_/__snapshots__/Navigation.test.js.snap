// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Interactions test will test user interaction 1`] = `
<div
  className="mx-5 my-3 pb-3 hstack gap-3"
  style={
    Object {
      "borderBottom": "1px solid black",
    }
  }
>
  <img
    alt="logo"
    src="logo.png"
    style={
      Object {
        "height": 65,
        "width": 65,
      }
    }
  />
  <h1>
    Space Travelers\` Hub
  </h1>
  <div
    className="ms-auto nav"
    onKeyDown={[Function]}
  >
    <div
      className="nav-item"
    >
      <a
        aria-current="page"
        className="nav-link-effect nav-link active"
        data-rr-ui-event-key={null}
        disabled={false}
        href="/"
        onClick={[Function]}
      >
        Rockets
      </a>
    </div>
    <div
      className="nav-item"
    >
      <a
        className="nav-link-effect nav-link"
        data-rr-ui-event-key={null}
        disabled={false}
        href="/missions"
        onClick={[Function]}
        style={
          Object {
            "marginRight": "1rem",
          }
        }
      >
        Missions
      </a>
    </div>
    <h6
      style={
        Object {
          "marginRight": "1.5rem",
          "marginTop": "0.5rem",
        }
      }
    >
      |
    </h6>
    <div
      className="nav-item"
    >
      <a
        className="nav-link-effect nav-link"
        data-rr-ui-event-key={null}
        disabled={false}
        href="/myprofile"
        onClick={[Function]}
        style={
          Object {
            "marginRight": "3.5rem",
          }
        }
      >
        My Profile
      </a>
    </div>
  </div>
</div>
`;
